app/src/main/kotlin/org/fossify/filemanager/activities/SimpleActivity.kt

```diff
@@ -31,10 +37,28 @@ open class SimpleActivity : BaseSimpleActivity() {
         R.mipmap.ic_launcher_grey_black
     )
 
+    companion object {
+        private const val MANAGE_STORAGE_RC = 201
+    }
+
     override fun getAppLauncherName() = getString(R.string.app_launcher_name)
 
     override fun getRepositoryName() = "File-Manager"
 
+    @SuppressLint("NewApi")
+    override fun onActivityResult(requestCode: Int, resultCode: Int, dataIntent: Intent?) {
+        super.onActivityResult(requestCode, resultCode, dataIntent)
+
+        if (requestCode == MANAGE_STORAGE_RC && isRPlus()) {
+            if (Environment.isExternalStorageManager()) {
+                recreate()
+            } else {
+                toast(R.string.no_storage_permissions)
+                finish()
+            }
+        }
+    }
+
```

Comment:
This should match the original code removed from MainActivity.

```suggestion
   @SuppressLint("NewApi")
    override fun onActivityResult(requestCode: Int, resultCode: Int, resultData: Intent?) {
        super.onActivityResult(requestCode, resultCode, resultData)
        isAskingPermissions = false
        if (requestCode == MANAGE_STORAGE_RC && isRPlus()) {
            actionOnPermission?.invoke(Environment.isExternalStorageManager())
        }
    }

```